src/api-service/__app__/onefuzzlib/azure/storage.py

```diff
@@ -0,0 +1,101 @@
+import logging
+import os
+import random
+from enum import Enum
+from typing import List, Tuple
+
+from azure.mgmt.storage import StorageManagementClient
+from memoization import cached
+from msrestazure.tools import parse_resource_id
+
+from .creds import get_base_resource_group, mgmt_client_factory
+
+
+class StorageType(Enum):
+    corpus = "corpus"
+    config = "config"
+
+
+@cached
+def get_fuzz_storage() -> str:
+    return os.environ["ONEFUZZ_DATA_STORAGE"]
+
+
+@cached
+def get_func_storage() -> str:
+    return os.environ["ONEFUZZ_FUNC_STORAGE"]
+
+
+@cached
+def get_primary_account(storage_type: StorageType) -> str:
+    if storage_type == StorageType.corpus:
+        # see #322 for discussion about typing
+        return get_fuzz_storage()
+    elif storage_type == StorageType.config:
+        # see #322 for discussion about typing
+        return get_func_storage()
+    raise NotImplementedError
+
+
+@cached
+def get_accounts(storage_type: StorageType) -> List[str]:
+    if storage_type == StorageType.corpus:
+        return corpus_accounts()
+    elif storage_type == StorageType.config:
+        return [get_func_storage()]
+    else:
+        raise NotImplementedError
+
+
+@cached
+def get_storage_account_name_key(account_id: str) -> Tuple[str, str]:
+    client = mgmt_client_factory(StorageManagementClient)
+    resource = parse_resource_id(account_id)
+    key = (
+        client.storage_accounts.list_keys(resource["resource_group"], resource["name"])
+        .keys[0]
+        .value
+    )
+    return resource["name"], key
+
+
+def choose_account(storage_type: StorageType) -> str:
+    accounts = get_accounts(storage_type)
+    if not accounts:
+        raise Exception(f"no storage accounts for {storage_type}")
+
+    if len(accounts) == 1:
+        return accounts[0]
+
+    # Use a random secondary storage account if any are available
```

Comment:
This means we _never_ choose a primary account if secondaries exist, right? Why is that desirable?

Reply:
The primary account account provides both containers as well as queues.  Secondary accounts only support containers, but at significantly higher performance rates.

Offloading container use to only secondary accounts reduces  IOP contention for the queues.

Reply:
Ah, that's a very good reason! Could we document it here (or elsewhere, if more appropriate).